.github/workflows/ci-pyright.yml

```diff
@@ -0,0 +1,36 @@
+# This workflow will install Python dependencies, run tests and lint with a single version of Python
```

Comment:
can drop single version of Python

Reply:
Should we keep only `3.10`?
That way, we are sure we don't use features only available from `3.11`.